pyo3-macros-backend/src/pyclass.rs

```diff
@@ -497,18 +506,21 @@ fn impl_descriptors(
         .flat_map(|(field, fns)| {
             fns.iter()
                 .map(|desc| {
-                    let name = field.ident.as_ref().unwrap().unraw();
-                    let doc = utils::get_doc(&field.attrs, None, true)
-                        .unwrap_or_else(|_| syn::LitStr::new(&name.to_string(), name.span()));
-                    let property_type = PropertyType::Descriptor(&field);
-                    match desc {
-                        FnType::Getter(self_ty) => {
-                            impl_py_getter_def(cls, property_type, self_ty, &name, &doc)
-                        }
-                        FnType::Setter(self_ty) => {
-                            impl_py_setter_def(cls, property_type, self_ty, &name, &doc)
+                    if let Some(name) = field.ident.as_ref().map(|ident| ident.unraw()) {
+                        let doc = utils::get_doc(&field.attrs, None, true)
+                            .unwrap_or_else(|_| syn::LitStr::new(&name.to_string(), name.span()));
+                        let property_type = PropertyType::Descriptor(&field);
+                        match desc {
+                            FnType::Getter(self_ty) => {
+                                impl_py_getter_def(cls, property_type, self_ty, &name, &doc)
+                            }
+                            FnType::Setter(self_ty) => {
+                                impl_py_setter_def(cls, property_type, self_ty, &name, &doc)
+                            }
+                            _ => unreachable!(),
                         }
-                        _ => unreachable!(),
+                    } else {
+                        bail_spanned!(field.span() => "get/set are not supported on tuple struct field");
```

Comment:
I'm curious what the reason for disallowing `get/set` is? I was thinking that it should be possible, but a `name` would be required.

e.g. this seems a plausible combination to me:

```rust
#[pyclass]
struct PyFoo(
    #[pyo3(get, set, name = "inner", from_py_with = "pytimestamp_to_int")] i32,
);
```

I'm happy for us to create an issue to leave this for a follow-up PR.

Reply:
It was disallowed just to avoid proc-maco panic when user try to use `get/set` on it. We can certainly add support for it, I think we can do the follow-up after #1507 merged.

Reply:
Hi @davidhewitt  and @messense  This's nice!  And as a local wrapper of external structure, I think we still can't get them work with nested getter and setter, which is `get_all` and `set_all`?